### PR TITLE
[fix] coverity issues

### PR DESCRIPTION
**Pull request by Rechi**
## Description
Fix a few coverity issues

## Motivation and Context
reduce coverity issues

## How Has This Been Tested?
build for windows and osx

## Types of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Improvement (non-breaking change which improves existing functionality)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the [Code guidelines](https://codedocs.xyz/xbmc/xbmc/code_guidelines.html) of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/xbmc/blob/master/CONTRIBUTING.md) document
- [ ] I have added tests to cover my change
- [x] All new and existing tests passed

### CODE REVIEW

File: xbmc/cores/AudioEngine/Utils/AEUtil.cpp [hunk 1]
```diff
@@ -554,12 +554,7 @@ AVSampleFormat CAEUtil::GetAVSampleFormat(AEDataFormat format)
     case AEDataFormat::AE_FMT_RAW:
       return AV_SAMPLE_FMT_U8;
     default:
-    {
-      if (AE_IS_PLANAR(format))
```
Comment (FernetMenta) [inline xbmc/cores/AudioEngine/Utils/AEUtil.cpp]: dropping this looks wrong on first glance. please elaborate

Comment (Rechi) [inline xbmc/cores/AudioEngine/Utils/AEUtil.cpp]: `AE_IS_PLANAR = AE_FMT_U8P|AE_FMT_S16NEP|AE_FMT_S32NEP|AE_FMT_S24NE4P|AE_FMT_S24NE4MSBP|AE_FMT_S24NE3P|AE_FMT_DOUBLEP|AE_FMT_FLOATP` and all those are already handled explicitly above.

Comment (FernetMenta) [inline xbmc/cores/AudioEngine/Utils/AEUtil.cpp]: if you argue like this, you need to remove entire default but this is not what we want to achive with the default case.
coverity is wrong here.

Comment (Rechi) [inline xbmc/cores/AudioEngine/Utils/AEUtil.cpp]: what should I do, drop this change?

File: xbmc/Util.cpp [hunk 2]
```diff
@@ -2352,14 +2352,14 @@ bool CUtil::ValidatePort(int port)
 
 int CUtil::GetRandomNumber()
```
Comment (Paxxi) [inline xbmc/Util.cpp]: This method should be removed or switched to using a proper C11 rand

File: xbmc/threads/Thread.h [hunk 3]
```diff
@@ -124,7 +124,7 @@ class CThread
   // -----------------------------------------------------------------------------------
 
   ThreadIdentifier m_ThreadId;
-  ThreadOpaque m_ThreadOpaque;
+  ThreadOpaque m_ThreadOpaque = { 0 };
```
Comment (Paxxi) [inline xbmc/threads/Thread.h]: This isn't entirely correct. On windows this is a HANDLE which is a pointer type

File: xbmc/video/Teletext.cpp [hunk 4]
```diff
@@ -32,21 +32,20 @@
...
-	}					\
-} while(0)
+static inline void SDL_memset4(void* dst, uint32_t val, size_t len)
+{
+  size_t _n = (len + 3) / 4;
+  uint32_t *_p = static_cast<uint32_t *>(dst);
+  if (len == 0)
+    return;
+  switch (len % 4) {
+    case 0: do { *_p++ = val;
```
Comment (Paxxi) [inline xbmc/video/Teletext.cpp]: This is completely broken. What is an unterminated do clause doing here?

**Review by Paxxi — CHANGES_REQUESTED** [review]
Sdl memset needs fixing rest looks good

File: xbmc/cores/AudioEngine/Utils/AEUtil.cpp (hunk 1 above)
Comment (FernetMenta) [inline xbmc/cores/AudioEngine/Utils/AEUtil.cpp]: yes, please

File: xbmc/video/Teletext.cpp (hunk 4 above)
Comment (Rechi) [inline xbmc/video/Teletext.cpp]: `} while ( --_n );` is a few lines below
I just replaced the define with a function so behaviour hasn't changed.
The define is an exact copy from libsdl.

Comment (garbear) [inline xbmc/video/Teletext.cpp]: This is known as Duff's device: https://en.wikipedia.org/wiki/Duff%27s_device

**Comment by notspiff** [comment]
@Rechi please consider if you want to do something about
```
1037: [/home/akva/kode/kodi-cmake/xbmc/threads/platform/win/ThreadLocal.h:45]: (warning) Class ThreadLocal is not safe, destructor throws exception
```

File: xbmc/video/Teletext.cpp (hunk 4 above)
Comment (Paxxi) [inline xbmc/video/Teletext.cpp]: Thanks garbear. Leave loop unrolling to the compiler I say. This code looks horrible

**Review by Paxxi — APPROVED** [review]

**Pull request merged**